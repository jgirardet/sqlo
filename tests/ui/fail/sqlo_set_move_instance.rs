#[derive(sqlo::Sqlo)]
struct Maison {
    id: i64,
    adresse: String,
    taille: i64,
    piscine: Option<bool>,
}

#[async_std::main]
async fn main() {
    let pool = sqlx::SqlitePool::connect(&std::env::var("DATABASE_URL").unwrap())
        .await
        .unwrap();

    let m = Maison::get(&pool, 1).await.unwrap();
    set_Maison![&pool, m, taille = 3].await.unwrap();
    m;
}
