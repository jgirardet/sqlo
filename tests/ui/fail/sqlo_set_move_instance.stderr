error[E0382]: use of moved value: `m`
  --> tests/ui/fail/sqlo_set_move_instance.rs:17:5
   |
1  | #[derive(sqlo::Sqlo)]
   |          ---------- value moved into closure here
...
15 |     let m = Maison::get(&pool, 1).await.unwrap();
   |         - move occurs because `m` has type `Maison`, which does not implement the `Copy` trait
16 |     update_Maison![m; taille = 3](&pool).await.unwrap();
   |                    - variable moved due to use in closure
17 |     m;
   |     ^ value used here after move
